warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/facet/FacetHandler.java

```diff
@@ -0,0 +1,363 @@
+        facetHashThreshold = conf.getInt(type.typeName() + FACET_HASH_THRESHOLD, 20);
+    }
+    
+    @Override
+    public String[] getTableNames(Configuration conf) {
+        final List<String> tableNames = new ArrayList<>();
+        
+        String tableName = conf.get(FACET_TABLE_NAME, null);
+        if (null != tableName)
+            tableNames.add(tableName);
+        
+        tableName = conf.get(FACET_METADATA_TABLE_NAME, null);
+        if (null != tableName)
+            tableNames.add(tableName);
+        
+        tableName = conf.get(FACET_HASH_TABLE_NAME, null);
+        if (null != tableName)
+            tableNames.add(tableName);
+        
+        return tableNames.toArray(new String[0]);
+    }
+    
+    @Override
+    public int[] getTableLoaderPriorities(Configuration conf) {
+        int[] priorities = new int[2];
+        int index = 0;
+        String tableName = conf.get(FACET_TABLE_NAME, null);
+        if (null != tableName)
+            priorities[index++] = conf.getInt(FACET_LPRIORITY, 40);
+        
+        tableName = conf.get(FACET_METADATA_TABLE_NAME, null);
+        if (null != tableName)
+            priorities[index++] = conf.getInt(FACET_METADATA_LPRIORITY, 40);
+        
+        tableName = conf.get(FACET_HASH_TABLE_NAME, null);
+        if (null != tableName)
+            priorities[index++] = conf.getInt(FACET_HASH_TABLE_LPRIORITY, 40);
+        
+        if (index != priorities.length) {
+            return Arrays.copyOf(priorities, index);
+        } else {
+            return priorities;
+        }
+    }
+    
+    @Override
+    public Multimap<BulkIngestKey,Value> processBulk(KEYIN key, RawRecordContainer event, Multimap<String,NormalizedContentInterface> fields,
+                    StatusReporter reporter) {
+        throw new UnsupportedOperationException("processBulk is not supported, please use process");
+    }
+    
+    @Override
+    public IngestHelperInterface getHelper(Type datatype) {
+        return datatype.getIngestHelper(this.taskAttemptContext.getConfiguration());
+    }
+    
+    @Override
+    public void close(TaskAttemptContext context) {
+        /* no-op */
+    }
+    
+    @Override
+    public RawRecordMetadata getMetadata() {
+        return null;
+    }
+    
+    protected byte[] flatten(ColumnVisibility vis) {
+        return markingFunctions == null ? vis.flatten() : markingFunctions.flatten(vis);
+    }
+    
+    @Override
+    public long process(KEYIN key, RawRecordContainer event, Multimap<String,NormalizedContentInterface> fields,
+                    TaskInputOutputContext<KEYIN,? extends RawRecordContainer,KEYOUT,VALUEOUT> context, ContextWriter<KEYOUT,VALUEOUT> contextWriter)
+                    throws IOException, InterruptedException {
+        
+        final String shardId = shardIdFactory.getShardId(event).substring(0, 8);
+        Text dateColumnQualifier = new Text(shardId); // TODO: Makes an assumption about the structure of the shardId.
```

Comment:
use ShardIdFactory.getDateString(shardId)

Reply:
final String shardId = ShardIdFactory.getDateString(shardId);

Reply:
Done. Thanks for the pointer to how to do this correctly.